detection_rules/rule.py

```diff
@@ -186,11 +186,12 @@ class RelatedIntegrations:
     filters: Optional[List[dict]]
     # trailing `_` required since `from` is a reserved word in python
     from_: Optional[str] = field(metadata=dict(data_key="from"))
-
+    history_window_start: Optional[str] = field(metadata=dict(metadata=dict(min_compat="8.4")))
```

Comment:
These 2 new fields should not go in the base class, but should create a new rule class `NewTermRuleData` that inherits `QueryRuleData`, since these fields are only valid in that rule type.

Reply:
Then, rather than restricting only these fields, we can restrict the entire rule type

Reply:
Re-factored the code to address both of these comments.